File: support/common/pgplot/sys_osf1/grgmem.c
#include <stdio.h>
#include <stdlib.h>
/*
  Fortran callable memory allocator

  Called as :
	ier = grgmem (size,pointer)

  where : size is an integer size of memory to allocate
	  pointer is an integer to return the pointer into

*/

#ifdef PG_PPU
#define GRGMEM grgmem_
#define GRFMEM grfmem_
#else
#define GRGMEM grgmem
#define GRFMEM grfmem
#endif

int GRGMEM(size, pointer)
int  *size;
#if 0
int *pointer;
#else
char **pointer;
#endif
{
#if 0
  char *area = malloc(*size);
  *pointer = (int)area;
  if (area == NULL) return 0;
  return 1;
#else
  if (!(*pointer=malloc(*size)))
      return 0;
  else
      return 1;
#endif
}

/*
  Fortran callable memory deallocator

  Called as :
	ier = grfmem (size,pointer)

  where : size is an integer size of memory to deallocate (not used)
	  pointer is an integer that contains the pointer

*/

int GRFMEM(size, pointer)
int *size;
#if 0
int *pointer;
#else
char **pointer;
#endif
{
#if 0
  char *area = (char *)*pointer;
  free(area);
  return 1;
#else
  free(*pointer);
  return 1;
#endif
}

File: branches/split_tools_and_core/mccode-legacy-tools/support/common/pgplot/sys_mac/grgmem.f
!!G Toolbox.finc
C  If you have a power mac version of LS fortran uncomment the
C next 5 lines and comment out "!!MP InLines.f"
C!!IFC NOT LSPOWERF
C!!MP 68KInlines
C!!ELSEC
C!!MP PPCInlines
C!!ENDC
C  If you have a 68K mac version of LS fortran comment out the 
C 5 lines above and uncomment the next line.
!!MP InLines.f
C  Fortran callable memory allocator

C  Called as :
C	ier = grgmem (size,pointer)

C  where : size is an integer size of memory to allocate
C	  pointer is an integer to return the pointer into

      Integer Function GRGMEM(size, pointer)
      Integer*4 size, pointer
      
      pointer = NewPtr(Size)
	  If (pointer .eq. 0) Then
	     grgmem = 0
	  Else
	     grgmem = 1
	  End if
      Return
	  End
	  
C  Fortran callable memory deallocator

C  Called as :
C	ier = grfmem (size,pointer)

C  where : size is an integer size of memory to deallocate (not used)
C	  pointer is an integer that contains the pointer


      Integer Function GRFMEM(size, pointer)
      Integer*4 size, pointer

      Call DisposPtr(Pointer)
	  grfmem = 1
	  Return
      End